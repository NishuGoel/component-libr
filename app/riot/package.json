{
  "name": "@storybook/riot",
  "version": "5.2.0-alpha.44",
  "description": "Storybook for riot.js: View riot snippets in isolation with Hot Reloading.",
  "keywords": [
    "storybook"
  ],
  "homepage": "https://github.com/storybookjs/storybook/tree/master/app/riot",
  "bugs": {
    "url": "https://github.com/storybookjs/storybook/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/storybookjs/storybook.git",
    "directory": "app/riot"
  },
  "license": "MIT",
  "main": "dist/client/index.js",
  "bin": {
    "build-storybook": "./bin/build.js",
    "start-storybook": "./bin/index.js",
    "storybook-server": "./bin/index.js"
  },
  "scripts": {
    "prepare": "node ../../scripts/prepare.js"
  },
  "dependencies": {
    "@storybook/core": "5.2.0-alpha.44",
    "common-tags": "^1.8.0",
    "core-js": "^3.0.1",
    "global": "^4.3.2",
    "raw-loader": "^2.0.0",
    "regenerator-runtime": "^0.12.1"
  },
  "devDependencies": {
    "@babel/plugin-transform-modules-commonjs": "^7.2.0",
    "@babel/preset-env": "^7.4.1",
    "@babel/preset-flow": "^7.0.0",
    "@babel/preset-react": "^7.0.0"
  },
  "peerDependencies": {
    "babel-loader": "^7.0.0 || ^8.0.0",
    "riot": "^3.0.0 || ^4.0.0",
    "riot-compiler": "^3.5.1 || ^4.0.0",
    "riot-hot-reload": "^1.0.0",
    "riot-tag-loader": "^2.0.0 || ^3.0.0"
  },
  "engines": {
    "node": ">=8.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
