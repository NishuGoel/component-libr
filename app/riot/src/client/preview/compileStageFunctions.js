import compiler from 'riot-compiler';

export const alreadyCompiledMarker = "var riot = require('riot')";

export function asCompiledCode(text) {
  return compiler
    .compile(text, {})
    .replace('var riot = require("riot")', '')
    .replace('riot.tag2(', 'tag2(');
}

export function compileNow(tag2, text) {
  // eslint-disable-next-line no-eval
  return tag2 && eval(asCompiledCode(text));
}

export function getRidOfRiotNoise(compiled) {
  return compiled.replace(/riot\.tag2/g, 'tag2').replace(alreadyCompiledMarker, '');
}

export function setConstructor(compiledSourceCode, constructor) {
  return compiledSourceCode.replace(/function\(opts\)\s*{\s*}(?=\);$)/, constructor.toString());
}
