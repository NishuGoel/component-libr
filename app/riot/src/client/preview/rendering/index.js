import renderCompiledButUnmounted from './compiledButUnmounted';
import renderStringified from './stringified';
import renderRaw from './raw';

export function render(component) {
  if (typeof component === 'string') {
    renderRaw(component);
    return true;
  }
  const { tags } = component || {};
  if (Array.isArray(tags)) {
    renderStringified(component);
    return true;
  }
  if (component && component.tagName) {
    renderCompiledButUnmounted(component);
    return true;
  }
  if (component && component.length) {
    // already rendered, nothing to do
    return true;
  }
  return false;
}
