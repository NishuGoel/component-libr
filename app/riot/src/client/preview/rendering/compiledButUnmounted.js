import { mount } from 'riot';

export default function renderCompiledButUnmounted(component) {
  mount('root', component.tagName, component.opts || {});
}
