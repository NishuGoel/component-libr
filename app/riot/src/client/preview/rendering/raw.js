import { mount, tag2 as tag } from 'riot';
import compiler from 'riot-compiler';
import { alreadyCompiledMarker, getRidOfRiotNoise } from '../compileStageFunctions';

export default function renderRaw(sourceCode) {
  const tag2 = tag;
  // eslint-disable-next-line no-eval
  eval(
    getRidOfRiotNoise(
      `${compiler.compile(sourceCode.replace(alreadyCompiledMarker, '').trim(), {})}`
    )
  );
  mount('root', /tag2\s*\(\s*'([^']+)'/.exec(sourceCode)[1], {});
}
