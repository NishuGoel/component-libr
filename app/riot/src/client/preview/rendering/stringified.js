/* eslint-disable import/no-duplicates */
import { mount, unregister, tag2 as tag } from 'riot';
import * as riot from 'riot';
import compiler from 'riot-compiler';
import { document } from 'global';
import { alreadyCompiledMarker, getRidOfRiotNoise, setConstructor } from '../compileStageFunctions';

function guessRootName(stringified) {
  const whiteSpaceLocation = stringified.indexOf(' ', stringified.indexOf('<') + 1);
  const firstWhitespace = whiteSpaceLocation === -1 ? stringified.length : whiteSpaceLocation;
  const supposedName = stringified.trim().match(/^<[^ >]+\/>$/)
    ? stringified.trim().replace(/[<>/]/g, '')
    : stringified.substring(
        stringified.indexOf('<') + 1,
        Math.min(firstWhitespace, stringified.indexOf('>'))
      );
  const matchingBuiltInTag = document.createElement(supposedName).constructor.name;
  return matchingBuiltInTag === 'HTMLUnknownElement' ? supposedName : 'root';
}

function compileText(code, rootName) {
  const sourceCodeEndOfHtml =
    (Math.min(code.indexOf('<style') + 1, code.indexOf('<script') + 1) || code.length + 1) - 1;
  const sourceCodeReformatted =
    code.substring(0, sourceCodeEndOfHtml).replace(/[\n\r\s]+/g, ' ') +
    code.substring(sourceCodeEndOfHtml);
  const sourceCode =
    rootName === 'root' ? `<root>${sourceCodeReformatted}</root>` : sourceCodeReformatted;
  return compiler
    .compile(sourceCode, {})
    .replace(alreadyCompiledMarker, '')
    .trim();
}

export default function renderStringified({
  tags,
  template = `<${(tags[0] || []).boundAs || guessRootName(tags[0] || '')}/>`,
  tagConstructor,
}) {
  const tag2 = tag;
  tags.forEach(oneTag => {
    const rootName = oneTag.boundAs || guessRootName(oneTag);
    const { content } = oneTag || {};
    const code = content ? content.trim() : oneTag || '';
    const compiled =
      code.indexOf(alreadyCompiledMarker) !== -1 ? code : compileText(code, rootName);
    unregister(rootName);
    eval(getRidOfRiotNoise(`${compiled}`)); // eslint-disable-line no-eval
  });
  const sourceCode = compiler.compile(`<root>${template}</root>`, {});

  let final;
  if (tagConstructor) {
    final = setConstructor(sourceCode, tagConstructor);
  } else {
    final = sourceCode;
  }

  if (template !== '<root/>') {
    eval(getRidOfRiotNoise(final)); // eslint-disable-line no-eval
  }

  mount('*');
}
