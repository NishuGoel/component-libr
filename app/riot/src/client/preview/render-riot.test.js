import { document } from 'global';
import { unregister, tag2, mount } from 'riot';
import compiler from 'riot-compiler';
import { render } from './rendering';
import { asCompiledCode } from './compileStageFunctions';

const rootElement = document.createElement('div');
rootElement.id = 'root';
document.body = document.createElement('body');
document.body.appendChild(rootElement);

const context = {
  unregister,
  compiler,
  tag2,
  mount,
};

beforeEach(() => {
  unregister('#root');
  rootElement.dataset.is = 'root';
});

describe('render a riot element', () => {
  it('should not work with nothing', () => {
    expect(render(null, context)).toBe(false);

    expect(rootElement.innerHTML).toEqual('');
  });

  it('can work with some text', () => {
    expect(render({ tags: ['<div><p>some tests</p></div>'] }, context)).toBe(true);

    expect(rootElement.innerHTML).toEqual('<div><p>some tests</p></div>');
  });

  it('can work with raw code', () => {
    expect(render("riot.tag2('root', '<div>raw code</div>', '', '', () => {})", context)).toBe(
      true
    );

    expect(rootElement.innerHTML).toEqual('<div>raw code</div>');
  });

  it('can work with compiled code', () => {
    expect(render([{}], context)).toBe(true);
    // does only work in true mode, and not in jest mode
  });

  it('will be possible to compile a template before rendering it', () => {
    const compiledTemplate = asCompiledCode('<template><div>raw code</div></template>');

    expect(compiledTemplate).toEqual(
      "tag2('template', '<div>raw code</div>', '', '', function(opts) {\n});"
    );
  });

  it('works with a json consisting in a tagName and opts', () => {
    tag2('hello', '<p>Hello { opts.suffix }</p>', '', '', () => {});

    expect(render({ tagName: 'hello', opts: { suffix: 'World' } }, context)).toBe(true);

    expect(rootElement.innerHTML).toEqual('<p>Hello World</p>');
  });

  it('can nest several tags', () => {
    expect(
      render(
        {
          tags: [
            '<Tag1><div>Inside tag1:<ul><li><Tag2><yield/></Tag2></li></ul></div></Tag1>',
            '<Tag2><div>Inside tag2:<ul><li><Tag3><yield/></Tag3></li></ul></div></Tag2>',
            '<Tag3><div>Inside tag3:<ul><li><Tag4><yield/></Tag4></li></ul></div></Tag3>',
            '<Tag4><div>Inside tag4:<ul><li><Tag5><yield/></Tag5></li></ul></div></Tag4>',
            '<Tag5><div>Inside tag5:<ul><li><yield/></li></ul></div></Tag5>',
          ],
          template: '<Matriochka><div><Tag1>Content</Tag1></div></Matriochka>',
        },
        context
      )
    ).toBe(true);

    expect(rootElement.innerHTML).toMatchSnapshot();
  });

  it('can template some vars', () => {
    expect(
      render(
        {
          tags: [
            {
              content:
                "<SimpleTest><div>simple test ({opts.test || 'without parameter'}). Oh, by the way ({opts.riotValue || '... well, nothing'})</div></SimpleTest>",
              boundAs: 'mustBeUniquePlease',
            },
          ],
          template:
            '<SimpleTest test={ "with a parameter" } value={"value is mapped to riotValue"}></SimpleTest>',
        },
        context
      )
    ).toBe(true);

    expect(rootElement.innerHTML).toMatchSnapshot();
  });

  it('can accept a constructor', () => {
    expect(
      render(
        {
          tags: [
            {
              content:
                "<SimpleTest><div>HACKED : {opts.hacked} ; simple test ({opts.test || 'without parameter'}). Oh, by the way ({opts.riotValue || '... well, nothing'})</div></SimpleTest>",
              boundAs: 'mustBeUniquePlease',
            },
          ],
          template:
            '<SimpleTest hacked={hacked} test={ "with a parameter" } value={"value is mapped to riotValue"}></SimpleTest>',
          tagConstructor: function tagConstructor() {
            this.hacked = true;
          },
        },
        context
      )
    ).toBe(true);

    expect(rootElement.innerHTML).toMatchSnapshot();
  });
});
