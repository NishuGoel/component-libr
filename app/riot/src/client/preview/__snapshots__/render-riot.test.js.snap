// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a riot element can accept a constructor 1`] = `"<simpletest hacked=\\"true\\" test=\\"with a parameter\\"><div>HACKED : true ; simple test (with a parameter). Oh, by the way (value is mapped to riotValue)</div></simpletest>"`;

exports[`render a riot element can nest several tags 1`] = `"<matriochka><div><tag1><div>Inside tag1:<ul><li><tag2><div>Inside tag2:<ul><li><tag3><div>Inside tag3:<ul><li><tag4><div>Inside tag4:<ul><li><tag5><div>Inside tag5:<ul><li>Content</li></ul></div></tag5></li></ul></div></tag4></li></ul></div></tag3></li></ul></div></tag2></li></ul></div></tag1></div></matriochka>"`;

exports[`render a riot element can template some vars 1`] = `"<simpletest test=\\"with a parameter\\"><div>simple test (with a parameter). Oh, by the way (value is mapped to riotValue)</div></simpletest>"`;
